bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/util/RaDecFields.scala

```diff
@@ -23,9 +24,23 @@ class DegreeFormatter(caption: String, inRange: Option[Double => Boolean] = None
   }
 }
 
+class DecDegreeFormatter(caption: String, inRange: Option[Double => Boolean] = None) extends Formatter[Declination](caption) {
+  val raDecFormat = new DecimalFormat("##0.000###")
+  def toString(d: Declination) = raDecFormat.format(d.toDegrees)
+
+  def fromString(s: String):Option[Declination] = try {
+    s.toDouble match {
+      case d if inRange.forall(f => f(d)) => Declination.fromAngle(Angle.fromDegrees(d))
+      case _                              => None
+    }
+  } catch {
+    case e:Exception             => None
+  }
+}
```

Comment:
I think all of this can be simplified using `Try` instead:
`def fromString(s: String): Option[Declination] = Try {
  s.toDouble match {
    case d if inRange.forall(f => f(d)) => Declination.fromAngle(Angle.fromDegrees(d))
}}.toOption`

should do the trick, I think.